### PR TITLE
dump memory usage for multiple modules

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
## What have you changed? (mandatory)

tikv-ctl support dump TiKV memory usage in multiple modules.

## What are the type of the changes? (mandatory)

- Misc (other changes)

## How has this PR been tested? (mandatory)

Manually, the output is like that:
```
 ./bin/tikv-ctl --host 172.16.5.37:20996 dump-memory-info
raftstore peers total memory usage 1888128, raftstore max peer memory usage 90024
StorageMemoryInfo { latch_mem_info: LatchMemoryInfo { total_mem_usage: 117440512, total_waiting: 9675, largest_slot_len: 2, largest_slot_capacity: 7 } }
```

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

File: src/storage/txn/latch.rs [hunk 1]
```diff
@@ -135,11 +135,11 @@ impl Latches {
             }
             // For some hot keys, the waiting list maybe very long, so we should shrink the waiting
             // VecDeque after pop.
-            if latch.waiting.capacity() > WAITING_LIST_SHRINK_SIZE
-                && latch.waiting.len() < WAITING_LIST_SHRINK_SIZE
-            {
-                latch.waiting.shrink_to_fit();
-            }
+            //            if latch.waiting.capacity() > WAITING_LIST_SHRINK_SIZE
```
Comment (siddontang) [inline src/storage/txn/latch.rs]: em, why comment these out?

**Comment by zhouqiang-cl** [comment]
It last too long, will be closed

**Pull request closed without merging**